desktop/core/src/desktop/api2.py

```diff
@@ -688,6 +692,155 @@ def share_document(request):
+  # Extract the task name and params from the request
+  try:
+    data = json.loads(request.body)
+    task_name = data.get('taskName')
+    task_params = data.get('taskParams')
+  except json.JSONDecodeError as e:
+    return JsonResponse({'error': str(e)}, status=500)
+
+  if task_name == 'document cleanup':
+    keep_days = task_params.get('keep-days')
+    from filebrowser.tasks import document_cleanup_task
+    task_kwargs = {
+      'keep_days': keep_days,
+      'user_id': request.user.id,
+      'username': request.user.username,
+    }
+    # Enqueue the document cleanup task with keyword arguments
+    task = document_cleanup_task.apply_async(kwargs=task_kwargs)
+
+    # Return a response indicating the task has been scheduled
+    return JsonResponse({
+      'taskName': task_name,
+      'taskParams': task_params,
+      'status': 'Scheduled',
+      'task_id': task.id,  # The task ID generated by Celery
+    })
+
+  elif task_name == 'tmp clean up':
+    from filebrowser.tasks import check_disk_usage_and_clean_task
+    cleanup_threshold = task_params.get('threshold for clean up')
+    disk_check_interval = task_params.get('disk check interval')
+    task_kwargs = {
+      'username': request.user.username,
+      'cleanup_threshold': cleanup_threshold,
+      'disk_check_interval': disk_check_interval
+    }
+    task = check_disk_usage_and_clean_task.apply_async(kwargs=task_kwargs)
+
+    return JsonResponse({
+      'taskName': task_name,
+      'status': 'Scheduled',
+      'task_id': task.id,  # The task ID generated by Celery
+    })
+
+  return JsonResponse({
+    'status': 0
+  })
+
+@api_error_handler
+#Retirve the tasks from the database
+def get_taskserver_tasks(request):
+
+  r = redis.Redis(host='localhost', port=6379, db=0)
+  tasks = []
+
+  # Use scan_iter to efficiently iterate over keys matching the first pattern
+  for key in r.scan_iter('celery-task-meta-*'):
+    task = json.loads(r.get(key))
+    tasks.append(task)
+
+  # Use scan_iter to efficiently iterate over keys matching the second pattern
+  for key in r.scan_iter('task:*'):
+    task = json.loads(r.get(key))
+    tasks.append(task)
+
+  return JsonResponse(tasks, safe=False)
+
+@api_error_handler
+def check_upload_status(request, task_id):
+  r = redis.Redis(host='localhost', port=6379, db=0)
+  task_key = f'celery-task-meta-{task_id}'
+  task_data = r.get(task_key)
+
+  if task_data is None:
+    return JsonResponse({'error': 'Task not found'}, status=404)
+
+  task = json.loads(task_data)
+  is_finalized = task.get('status') == 'SUCCESS'
+  is_running = task.get('status') == 'RUNNING'
+  is_failure = task.get('status') == 'FAILURE'
+  is_revoked = task.get('status') == 'REVOKED'
+
+  return JsonResponse({'isFinalized': is_finalized, 'isRunning': is_running, 'isFailure': is_failure, 'isRevoked': is_revoked})
+
+from django.http import JsonResponse
+from celery.app.control import Control
+from desktop.celery import app as celery_app
```

Comment:
Is there a reason to keep these imports here in between the file? Unless these is some dependency issue, the imports should be at the top. It helps with maintaining the imports.